emscripten.py

```diff
@@ -2176,8 +2179,24 @@ def emscript_wasm_backend(infile, outfile, memfile, compiler_engine,
   #   * Run wasm-emscripten-finalize to extract metadata and modify the binary
   #     to use emscripten's wasm<->JS ABI
   #   * Use the metadata to generate the JS glue that goes with the wasm
+  if shared.Settings.MAIN_MODULE:
+    # Run compile_settings here to retrieve all the JS library functions.
+    # Unfortunately, we need to run it later again to capture the exported functions
+    # from the Wasm
+    glue, forwarded_data = compile_settings(compiler_engine, temp_files)
+    forwarded_json = json.loads(forwarded_data)
+    library_fns = forwarded_json['Functions']['libraryFunctions']
+    library_fns_list = []
+    for name in library_fns:
+      library_fns_list.append(name.encode('utf-8'))
+
+    libraryfile = infile + '.libfile'
+    with open(libraryfile, 'w') as f:
+      f.write(str(library_fns_list))
```

Comment:
I better format would probably just one line per symbols:

```
for sym in library_fns_list:
   f.write(sym + '\n')
```

Reply:
Can we call the file, and this variable somehow more meaningful too? Perhaps "js_imports"?  Or "js_symbols"?

Reply:
Thanks for the suggestion, will update the format :)